File: frontend/pusgs-front-app/src/app/model/order.ts
export interface OrderProduct {
  productName: string;
  price: number;
  ingredients: string;
  count: number;
}

export class Order {
  deliveryAddress: string = '';
  comment: string = '';
  price: number = 0;
  orderState: number = 0;
  lat: number = 0;
  lon: number = 0;
}

export class OrderWithId {
  id: number = -1;
  deliveryAddress: string = '';
  comment: string = '';
  price: number = 0;
  orderState: number = 0;
  lat: number = 0;
  lon: number = 0;
  product: string = '';
}

File: backend/DeliveryApplication/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DeliveryApplication.Models
{
    public enum OrderStateEnum { PENDING , ACTIVE , FINISHED  };
    public class Order
    {
        public int Id { get; set; }
        public string DeliveryAddress { get; set; }
        public string Comment { get; set; }
        public double Price { get; set; }
        public OrderStateEnum OrderState { get; set; }
        public virtual ICollection<OrderProduct> OrderProducts { get; set; }
        public User Deliverer { get; set; }
        public virtual ICollection<UserOrder> UserOrders { get; set; }
        public double Lat { get; set; }
        public double Lon { get; set; }
    }
}
